service/src/integrationTest/kotlin/fi/espoo/evaka/assistanceneed/decision/AssistanceNeedDecisionIntegrationTest.kt

```diff
@@ -763,6 +766,338 @@ class AssistanceNeedDecisionIntegrationTest : FullApplicationTest(resetDbBeforeE
+        val today = LocalDate.of(2022, 12, 31)
+        db.transaction { tx ->
+            tx.insert(
+                DevPlacement(
+                    type = PlacementType.DAYCARE,
+                    childId = testChild_1.id,
+                    unitId = testDaycare.id,
+                    startDate = startDate,
+                    endDate = endDate,
+                )
+            )
+        }
+        val decision =
+            createAssistanceNeedDecision(
+                AssistanceNeedDecisionRequest(
+                    testDecision.copy(
+                        validityPeriod = DateRange(startDate, null),
+                        selectedUnit = UnitIdInfo(testDaycare.id)
+                    )
+                )
+            )
+        sendAssistanceNeedDecision(decision.id)
+        decideAssistanceNeedDecision(
+            decision.id,
+            AssistanceNeedDecisionController.DecideAssistanceNeedDecisionRequest(
+                status = AssistanceNeedDecisionStatus.ACCEPTED
+            ),
+            decisionMaker
+        )
+
+        db.transaction { tx -> tx.endActiveDaycareAssistanceDecisions(today) }
+
+        assertEquals(
+            DateRange(startDate, null),
+            getAssistanceNeedDecision(decision.id).validityPeriod
+        )
+    }
+
+    @Test
+    fun `endActiveDaycareAssistanceDecisions should set end date if placement ended yesterday`() {
+        val startDate = LocalDate.of(2022, 1, 1)
+        val endDate = LocalDate.of(2022, 12, 31)
+        val today = LocalDate.of(2023, 1, 1)
+        db.transaction { tx ->
+            tx.insert(
+                DevPlacement(
+                    type = PlacementType.DAYCARE,
+                    childId = testChild_1.id,
+                    unitId = testDaycare.id,
+                    startDate = startDate,
+                    endDate = endDate,
+                )
+            )
+        }
+        val decision =
+            createAssistanceNeedDecision(
+                AssistanceNeedDecisionRequest(
+                    testDecision.copy(
+                        validityPeriod = DateRange(startDate, null),
+                        selectedUnit = UnitIdInfo(testDaycare.id)
+                    )
+                )
+            )
+        sendAssistanceNeedDecision(decision.id)
+        decideAssistanceNeedDecision(
+            decision.id,
+            AssistanceNeedDecisionController.DecideAssistanceNeedDecisionRequest(
+                status = AssistanceNeedDecisionStatus.ACCEPTED
+            ),
+            decisionMaker
+        )
+
+        db.transaction { tx -> tx.endActiveDaycareAssistanceDecisions(today) }
+
+        assertEquals(
+            DateRange(startDate, endDate),
+            getAssistanceNeedDecision(decision.id).validityPeriod
+        )
+    }
+
+    @Test
+    fun `endActiveDaycareAssistanceDecisions should not set end date if placement type changes to preschool daycare`() {
```

Comment:
Maybe add to description that the end date is set to the end of the last placement